web/src/main/java/com/epam/rft/atsy/web/controllers/rest/CandidateApplicationController.java

```diff
@@ -36,21 +38,24 @@
    * @return a collection with all applications of the specified candidate
    */
   @RequestMapping(method = RequestMethod.GET)
-  public Collection<CandidateApplicationDTO> loadApplications(
-      @PathVariable(value = "candidateId") Long candidateId, Locale locale) {
-    Collection<CandidateApplicationDTO>
-        applicationStates =
-        statesHistoryService
-            .getCandidateApplicationsByCandidateIdOrderByModificationDateDesc(candidateId);
-
-    for (CandidateApplicationDTO candidateApplicationDTO : applicationStates) {
+  public PagingResponse<CandidateApplicationDTO> loadApplications(
+      @PathVariable(value = "candidateId") Long candidateId, Locale locale,
+      @RequestParam(name = "pageNumber") Integer pageNumber,
+      @RequestParam(name = "pageSize") Integer pageSize) {
+
+    PagingResponse<CandidateApplicationDTO>
+        candidateApplicationDTOPagingResponse =
+        applicationsService.getApplicationsByCandidateId(candidateId, pageNumber - 1, pageSize);
+
+    for (CandidateApplicationDTO candidateApplicationDTO : candidateApplicationDTOPagingResponse
+        .getRows()) {
       String stateType = candidateApplicationDTO.getStateType();
 
       stateType =
           messageKeyResolver.resolveMessageOrDefault(APPLICATION_STATE + stateType, stateType);
 
       candidateApplicationDTO.setStateType(stateType);
```

Comment:
I know this is the original code, but there is no sense in creating a variable for this issue. There is an easier and more compact way, simply: candidateApplicationDTO.setStateType(messageKeyResolver.resolveMessageOrDefault(APPLICATION_STATE + stateType, stateType));. Could you please correct this code part?